apps/src/dance/DanceAiClient.ts

```diff
@@ -0,0 +1,100 @@
+import UntypedCachedPalettes from '@cdo/static/dance/ai/model/cached-spacy-palette-map.json';
+import UntypedCachedBackgrounds from '@cdo/static/dance/ai/model/cached-spacy-background-map.json';
+import UntypedCachedForegrounds from '@cdo/static/dance/ai/model/cached-spacy-foreground-map.json';
+
+// Custom types for vector mapping JSONs to play nice with Typescript
+type EmojiAssociation = {
+  association: number[];
+};
+
+type CachedMapping = {
+  emojiAssociations: {[key: string]: EmojiAssociation};
+  output: string[];
+};
+
+const CachedPalettes: CachedMapping = UntypedCachedPalettes;
+const CachedBackgrounds: CachedMapping = UntypedCachedBackgrounds;
+const CachedForegrounds: CachedMapping = UntypedCachedForegrounds;
+
+// Output: JSON containing {backgroundColor:string, backgroundEffect:string, foregroundEffect:string}
+// ToDo: Break down into helper functions for each step rather than repeat code for each field
+export default function cachedAi(input: string) {
+  const inputWords: string[] = input.split(',');
+  const trimmedInput: string[] = inputWords.map(value => value.trim());
+  const inputPaletteVectors: {[key: string]: any} = {};
+  const inputBackgroundVectors: {[key: string]: any} = {};
+  const inputForegroundVectors: {[key: string]: any} = {};
```

Comment:
I noticed that these caused lint errors when I tried to commit. Please fix all lint errors before merging.

Reply:
That's super weird, maybe it's because of the any? I have the VSCode auto-run the prettier lint whenever I save so I'm surprised it didn't catch that